subsys/bluetooth/controller/ll_sw/nordic/lll/lll_df.c

```diff
@@ -65,44 +67,60 @@ uint8_t lll_df_ant_num_get(void)
 	return radio_df_ant_num_get();
 }
 
-/* @brief Function initializes transmission of Constant Tone Extension.
- *
- * @param[in] type      Type of CTE: AoA, AoD 1us, AoD 2us
- * @param[in] length    Duration of CTE in 8us units
- * @param[in] ant_num   Number of antennas in switch pattern
- * @param[in] ant_ids   Antenna identifiers that create switch pattern.
+#if defined(CONFIG_BT_CTLR_DF_ADV_CTE_TX)
+/* @brief Function enables transmission of Constant Tone Extension.
  *
- * @Note Pay attention that first two antenna identifiers in a switch
- * pattern has special purpose. First one is used in guard period, second
- * in reference period. Actual switching is processed from third antenna.
- *
- * In case of AoA mode ant_num and ant_ids parameters are not used.
+ * @param lll_sync        Pointer to LLL sync. object associated with
+ *                        periodic advertising event.
+ * @param pdu             Pointer to PDU that will be transmitted.
+ * @param[out] cte_len_us Pointer to store actual CTE length in [us]
  */
-void lll_df_conf_cte_tx_enable(uint8_t type, uint8_t length,
-			       uint8_t ant_num, uint8_t *ant_ids)
+void lll_df_cte_tx_enable(struct lll_adv_sync *lll_sync, const struct pdu_adv *pdu,
+			  uint32_t *cte_len_us)
```

Comment:
What is the reasoning of providing a `uint32_t` out-pointer, rather than returning the value?

Reply:
I thought about returning the cte_len_us by `return` keyword, but in my humble opinion use of pointer is more meaningful. You have the name of the pointer that suggests what is the purpose. If I had seen call to `lll_df_cte_tx_enable` I would expect it returns error code by 'return` statement.

Reply:
> If I had seen call to lll_df_cte_tx_enable I would expect it returns error code by 'return` statement.

That is a fair point.